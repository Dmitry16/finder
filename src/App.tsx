import React from 'react';
import './App.css';

const App: React.FC = () => {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          Welcome to the awsome Dropdown App!
        </p>
      </header>
    </div>
  );
}

export default App;
