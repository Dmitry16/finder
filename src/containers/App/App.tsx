import React from 'react';
import Dropdown from '../../components/Dropdown';
import styled from 'styled-components';
import ErrorBoundary from '../ErrorBoundary';
import { flex, font, height } from '../../common/styles';
import { api } from '../../api';

const Header = styled.header`
  background-color: #282c34;
  ${height}
  ${font}
  ${flex}
  color: white;
`
const Section = styled.section`
  background-color: white;
  ${height}
  ${font}
  ${flex}
`
const Footer = styled.footer`
  background-color: #282c34;
  ${height}
  ${font}
  ${flex}
`
const Container = styled.div`
  text-align: center;
`

const App: React.FC = () => {

  const { data } = api;
  const dropdownInitialValue = data && data[0];

  const clickHandler = (value: string) => {
    console.log('clickHandler!!', value)
  }

  return (
    <ErrorBoundary>
      <Container>
        <Header>
          <p>
            Welcome to the awsome Finder App!
          </p>
        </Header>
        <Section>
          <Dropdown
            listBlockHeight={5}
            clickHandler={clickHandler}
            initialValue={dropdownInitialValue}
          />
        </Section>
        <Footer />
      </Container>
    </ErrorBoundary>
  );
}

export default App;
