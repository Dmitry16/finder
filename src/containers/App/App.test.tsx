import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import App from './App';

describe('<App />', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders determined layout', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('renders Dropdown component', () => {
    expect(wrapper.find('Dropdown').length).toEqual(1);
  });
  it('Dropdown component is been passed passedValue prop', () => {
    expect('passedValue' in wrapper.find('Dropdown').props()).toBe(true);
  });
  it('the passedValue prop is a string', () => {
    expect(typeof wrapper.find('Dropdown').prop('passedValue')).toBe('string');
  });
  it('Dropdown component is been passed clickHandler prop', () => {
    expect('clickHandler' in wrapper.find('Dropdown').props()).toBe(true);
  });
  it('the clickHandler prop is a function', () => {
    expect(wrapper.find('Dropdown').prop('clickHandler')).toBeInstanceOf(Function);
  });
  it('Dropdown component is been passed listBlockHeight prop', () => {
    expect('listBlockHeight' in wrapper.find('Dropdown').props()).toBe(true);
  });
  it('the listBlockHeight prop is a number', () => {
    expect(typeof wrapper.find('Dropdown').prop('listBlockHeight')).toBe('number');
  });
});

