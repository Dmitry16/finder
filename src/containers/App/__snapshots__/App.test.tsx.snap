// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders determined layout 1`] = `ShallowWrapper {}`;
